### PR TITLE
feat: isStrict option for no-unnecessary-act

### PR DESCRIPTION
**Pull request by zaicevas**
Closes https://github.com/testing-library/eslint-plugin-testing-library/issues/382

### CODE REVIEW

File: lib/rules/no-unnecessary-act.ts [hunk 1]
```diff
@@ -32,37 +33,71 @@ export default createTestingLibraryRule<[], MessageIds>({
         'Avoid wrapping Testing Library util calls in `act`',
       noUnnecessaryActEmptyFunction: 'Avoid wrapping empty function in `act`',
     },
-    schema: [],
+    schema: [
```
Comment (zaicevas) [inline lib/rules/no-unnecessary-act.ts:36]: Added it for consistency, but not sure why `schema` is needed and how and where it's used. Can someone enlighten me? 😄

Comment (Belco90) [reply lib/rules/no-unnecessary-act.ts:36]: You can find ESLint Options Schemas doc here: https://eslint.org/docs/developer-guide/working-with-rules#options-schemas

TL;DR ESLint will use that schema to validate configuration options and prevent invalid or unexpected inputs before  executing the rules:

ESLint config validation error displayed based on rule schema

File: docs/rules/no-unnecessary-act.md [hunk 2]
```diff
@@ -87,6 +87,30 @@ await act(async () => {
 });
 ```
 
+## Options
+
+This rule has one option:
+
+- `isStrict`: wrapping both things related and not related to Testing Library in `act` is reported
```
Comment (Belco90) [inline docs/rules/no-unnecessary-act.md:94]: Could you specify this is `false` by default?

File: lib/rules/no-unnecessary-act.ts (hunk 1 above)
Comment (Belco90) [reply lib/rules/no-unnecessary-act.ts:36]: I just realized the option for `no-dom-import` is specified just as a `string` in the schema, but can be improved to indicate only `"angular"`, `"react"`, or `"vue"` are the valid options.

File: lib/rules/no-unnecessary-act.ts [hunk 3]
```diff
@@ -89,7 +124,16 @@ export default createTestingLibraryRule<[], MessageIds>({
           node: identifierNode,
           messageId: 'noUnnecessaryActEmptyFunction',
         });
-      } else if (!hasSomeNonTestingLibraryCall(blockStatementNode.body)) {
+        return;
+      }
+
+      const isReported =
```
Comment (Belco90) [inline lib/rules/no-unnecessary-act.ts:130]: I found this var name confusing! I thought it meant the code violation was already reported, but it actually indicates if it needs to be reported. Could you rename it to something like `shouldBeReported`?

File: lib/rules/no-unnecessary-act.ts [hunk 4]
```diff
@@ -89,7 +124,16 @@ export default createTestingLibraryRule<[], MessageIds>({
...
+        (!isStrict && !hasSomeNonTestingLibraryCall(blockStatementNode.body)) ||
+        (isStrict &&
+          hasSomeNonTestingLibraryCall(blockStatementNode.body) &&
+          hasTestingLibraryCall(blockStatementNode.body));
```
Comment (Belco90) [inline lib/rules/no-unnecessary-act.ts:134]: I guess this can be simplified like this? What do you think?

```suggestion
const isReported = isStrict
  ? hasSomeNonTestingLibraryCall(blockStatementNode.body) &&
    hasTestingLibraryCall(blockStatementNode.body)
  : !hasSomeNonTestingLibraryCall(blockStatementNode.body);
```

**Review by Belco90 — CHANGES_REQUESTED** [review]
Thanks for your contribution again! This was a really quick one. Just some minor changes and should be ready.

**Review by Belco90 — APPROVED** [review]
Great work again! You are becoming a veteran of this codebase 😅
